tests/asdf_tests/test_core.py

```diff
@@ -0,0 +1,362 @@
+    )
+
+    with pytest.raises(ValidationError):
+        buff = BytesIO()
+        with asdf.AsdfFile(
+            {"lcs": lcs}, extensions=[WeldxExtension(), WeldxAsdfExtension()]
+        ) as f:
+            f.write_to(buff)
+
+
+# TODO: remove
+test_local_coordinate_system_shape_violation()
+
+
+# weldx.transformations.CoordinateSystemManager ----------------------------------------
+
+buffer_csm = BytesIO()
+
+
+def are_coordinate_system_managers_equal(
+    csm_0: tf.CoordinateSystemManager, csm_1: tf.CoordinateSystemManager
+):
+    """
+    Test if two CoordinateSystemManager instances are equal.
+
+    Parameters
+    ----------
+    csm_0:
+        First CoordinateSystemManager instance.
+    csm_1:
+        Second CoordinateSystemManager instance.
+
+    Returns
+    -------
+    bool:
+        True if both coordinate system managers are identical, False otherwise
+    """
+    graph_0 = csm_0.graph
+    graph_1 = csm_1.graph
+
+    if len(graph_0.nodes) != len(graph_1.nodes):
+        return False
+    if len(graph_0.edges) != len(graph_1.edges):
+        return False
+
+    # check nodes
+    for node in graph_0.nodes:
+        if node not in graph_1.nodes:
+            return False
+
+    # check edges
+    for edge in graph_0.edges:
+        if edge not in graph_1.edges:
+            return False
+
+    # check coordinate systems
+    for edge in graph_0.edges:
+        lcs_0 = csm_0.get_local_coordinate_system(edge[0], edge[1])
+        lcs_1 = csm_1.get_local_coordinate_system(edge[0], edge[1])
+        if not are_local_coordinate_systems_equal(lcs_0, lcs_1):
+            return False
+
+    return True
+
+
+def get_example_coordinate_system_manager():
+    """Get a consistent CoordinateSystemManager instance for test purposes."""
+    csm = tf.CoordinateSystemManager("root")
+    csm.create_cs("lcs_01", "root", coordinates=[1, 2, 3])
+    csm.create_cs(
+        "lcs_02",
+        "root",
+        orientation=tf.rotation_matrix_z(np.pi / 3),
+        coordinates=[4, -7, 8],
+    )
+    csm.create_cs(
+        "lcs_03",
+        "lcs_02",
+        orientation=tf.rotation_matrix_y(np.pi / 11),
+        coordinates=[4, -7, 8],
+    )
+    return csm
+
+
+def test_coordinate_system_manager_save():
+    """Test if a CoordinateSystemManager can be written to an asdf file."""
+    csm = get_example_coordinate_system_manager()
+    tree = {"cs_hierarchy": csm}
+    with asdf.AsdfFile(
+        tree, extensions=[WeldxExtension(), WeldxAsdfExtension()], copy_arrays=True
+    ) as f:
+        f.write_to(buffer_csm)
+        buffer_csm.seek(0)
+
+
+def test_coordinate_system_manager_load():
+    """Test if a CoordinateSystemManager can be read from an asdf file."""
+    f = asdf.open(buffer_csm, extensions=[WeldxExtension(), WeldxAsdfExtension()])
+    csm_exp = get_example_coordinate_system_manager()
+    csm_file = f.tree["cs_hierarchy"]
+
+    assert are_coordinate_system_managers_equal(csm_exp, csm_file)
```

Comment:
We should define a "save -> buffer -> load utility function" similar to  [`asdf.tests.helpers.assert_roundtrip_tree`](https://github.com/asdf-format/asdf/blob/master/asdf/tests/helpers.py#L154) but wiht io.BytesIO instead of files